lib/libc/stdio.c

```diff
@@ -72,10 +229,23 @@ int getchar(void) {
     return getc(stdin);
 }
 
+char *fgets(char *s, int size, FILE *stream) {
+    int c = -1;
+    char *cs = s;
+
```

Comment:
might be worth adding an error check for size < 1, since that should be impossible to fill in a string.